File: api/app/BusinessCard.php
<?php

namespace App;


use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\SoftDeletes;
use Illuminate\Support\Str;

class BusinessCard extends Model
{
    use SoftDeletes;
    protected $fillable = [
        'name', 'company_name', 'position', 'address', 'private',
        'mobile', 'email', 'website', 'image_path', 'created_by', 'deleted_by'
    ];

    protected $with = [
        'notes'
    ];

    protected $table = 'business_cards';

    public function created_by()
    {
        return $this->belongsTo(User::class, 'created_by');
    }

    public function permissions()
    {
        return $this->hasMany(PermissionUser::class, 'business_card_id');
    }

    public function notes() {
        return $this->hasMany(Note::class);
    }

    protected static function boot()
    {
        parent::boot();

        self::creating(function ($post) {
            $post->{$post->getKeyName()} = $post->{$post->getKeyName()} ?? Str::uuid();
        });
    }

    public function getKeyName()
    {
        return 'id';
    }

    public function getKeyType()
    {
        return 'string';
    }

    public function getIncrementing()
    {
        return false;
    }

}

File: android/app/src/main/java/com/ohmycthulhu/businesscarddatabase/BusinessCard.kt
package com.ohmycthulhu.businesscarddatabase

import java.io.Serializable

class BusinessCard (name: String, company: String?, email: String?, address: String?, phone: String?, website: String?, position: String?, private: Boolean, note: String?, imagePath: String?) : Serializable {
    var name: String
    var company: String?
    var position: String?
    var email: String?
    var address: String?
    var phone: String?
    var website: String?
    var private: Boolean
    var note: String?
    var imagePath: String?

    init {
        this.name = name
        this.company = company
        this.position = position
        this.email = email
        this.address = address
        this.phone = phone
        this.website = website
        this.private = private
        this.note = note
        this.imagePath = imagePath
    }

}